marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/mapping.sjs

```diff
@@ -86,6 +86,10 @@ function defaultArtifact(artifactName, entityTypeId) {
     defaultCollections.push(hubEs.findEntityServiceTitle(entityTypeId) || entityTypeId);
   }
   return {
+    headers: {
```

Comment:
I'm realizing now that while this will fix the problem for steps created starting in 5.4.0, it's still a problem for everyone's mapping steps prior to 5.4.0. In that sense, we're adding a new feature, but not fixing the bug. 

The general problem here is that when a DHF step is creating a new document based on an input document, it shouldn't blindly copy createdOn/createdBy from the input document. The value of createdOn is almost certainly incorrect, and createdBy may be incorrect too. 

I don't think this should be fixed by step config, this should be fixed by code. 

This appears to only be a problem with mapping steps, as those are the only ones calling the "mergeHeaders" function. I think we need a code fix in that function that doesn't depend on step config. 

That function should say - after I merge the headers, then if createdOn/createdBy are present, I'll be sure to update those to the actual values - fn.currentDateTime() and xdmp.currentUser().